### DISCUSSION TITLE
Event based incremental static Regeneration

### DISCUSSION THREAD

**Discussion by Andu033** [vercel/next.js#16903] [Help]
For now, I know that is no way provided by the framework to invalidate a static page when you want. The only way is to set an interval of time to revalidate the page every time. 

I'm working with some static pages that update their data rarely but when it does I want to ship the page with updates made. 
I tried a workaround to have an express server running on the same machine with and endpoint that call a function that deletes the desired file from the server. But the page is not generated immediately anyway. It takes a variating number of minutes to regenerate the page. Until then it just serves the old page from the memory cache probably. 

Is there any way of forcing the regeneration of a page? Has anyone done that?

**Comment by felipeix** [comment]
I also need this so much. Is this going to be implemented?
